js/paperhive.js

```diff
@@ -18,363 +18,367 @@
+
+        if (menuItem.length && menuItem.attr('data-action') === 'paperhive') {
+            OCA.Files_PaperHive.createNotification();
+        } else if (!notificationItem.length) {
+            OCA.Files_PaperHive.hideNotification();
+        }
+    }
 };
 
 Files_PaperHive.NewFileMenuPlugin = {
 
-	attach: function(menu) {
-		var fileList = menu.fileList;
-
-		// only attach to main file list, public view is not supported yet
-		if (fileList.id !== 'files') {
-			return;
-		}
-
-		// register the new menu entry
-		menu.addMenuEntry({
-			id: 'paperhive',
-			displayName: t('files_paperhive', 'PaperHive Book'),
-			templateName: t('files_paperhive', 'Your Book ID'),
-			iconClass: 'icon-filetype-paperhive',
-			fileType: 'file',
-			actionHandler: function(bookID) {
-				// Hide any remaining notification
-				OCA.Files_PaperHive.hideNotification();
-
-				// Get the file data from PaperHive API
-				var dir = fileList.getCurrentDirectory();
-				var $saveNot = OC.Notification.showHtml(t('files_paperhive', 'Saving..'));
-				OCA.Files_PaperHive.getPaperHiveBook(
-					dir,
-					bookID,
-					function(paperHiveData) {
-						fileList.addAndFetchFileInfo(paperHiveData.path, '', {scrollTo: true}).then(
-							function(status, data) {
-								var $tr = fileList.findFileEl(paperHiveData.filename);
-								OCA.Files_PaperHive._updatePaperHiveFileData($tr, paperHiveData.discussionCount);
-								OC.Notification.hide($saveNot);
-							},
-							function() {
-								OC.Notification.hide($saveNot);
-								OCA.Files_PaperHive.failureNotification('Could not create file');
-							}
-						);
-					},
-					function(message){
-						OC.Notification.hide($saveNot);
-						OCA.Files_PaperHive.failureNotification(message);
-					}
-				);
-			}
-		});
-	}
+    attach: function (menu) {
+        var fileList = menu.fileList;
+
+        // only attach to main file list, public view is not supported yet
+        if (fileList.id !== 'files') {
+            return;
+        }
+
+        // register the new menu entry
+        menu.addMenuEntry({
+            id: 'paperhive',
+            displayName: t('files_paperhive', 'PaperHive Book'),
+            templateName: t('files_paperhive', 'Your Book ID'),
+            iconClass: 'icon-filetype-paperhive',
+            fileType: 'file',
+            actionHandler: function (bookID) {
+                // Hide any remaining notification
+                OCA.Files_PaperHive.hideNotification();
+
+                // Get the file data from PaperHive API
+                var dir = fileList.getCurrentDirectory();
+                var $saveNot = OC.Notification.showHtml(t('files_paperhive', 'Saving...'));
+                OCA.Files_PaperHive.getPaperHiveBook(
+                    dir,
+                    bookID,
+                    function (paperHiveData) {
+                        var filename = paperHiveData.filename+paperHiveData.extension;
+
+                        fileList.filesClient.getFileInfo(
```

Comment:
Can I use filesClient in here to do PROPFIND for this file? I want to do this in order to add file into the list (can I get all the properties from PUT to serever?